ert3/config/_parameters_config.py

```diff
@@ -105,27 +105,53 @@ def __len__(self) -> int:
         return len(self.__root__)
 
 
+def _ensure_valid_size(size: int) -> Optional[int]:
+    if size is not None and size <= 0:
+        raise ValueError("Size cannot be <= 0")
+
+    return size
+
+
 class _ParameterConfig(_ParametersConfig):
     name: str
     type: Literal["stochastic"]
     distribution: Union[_GaussianDistribution, _UniformDistribution]
-    variables: _VariablesConfig
+    variables: Optional[_VariablesConfig] = None
+    size: Optional[int] = None
+
+    @root_validator
+    def _ensure_variables_or_size(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if (values.get("variables") is None and values.get("size") is not None) or (
```

Comment:
What if both are `None`? 
Could use xor here:
```suggestion
        if  (values.get("variables") is None) ^ (values.get("size") is None):
            return values
        raise AssertionError("Parameter group must contain either variables or size")

```

Reply:
I don't find `xor` to be particularly pythonic... Also, I think writing this in two statements gives better feedback to the users.

```suggestion
        has_variables = values.get("variables") is not None
        has_size = values.get("size") is not None
        if has_variables and has_size:
            raise AssertionError("A parameter group cannot have both size and variables specified")
        elif not has_variables and not has_size:
            raise AssertionError("A parameter group cannot have neither a size nor variables specified")
        else:
            return variables
```

Reply:
Both being `None` will raise an AssertionError, but I agree that separate error messages give better feedback.